src/platform/vita/vita_touch.c

```diff
@@ -0,0 +1,343 @@
+    int last_x; // last known screen coordinates
+    int last_y; // last known screen coordinates
+    float last_down_x; // SDL touch coordinates when last pressed down
+    float last_down_y; // SDL touch coordinates when last pressed down
+} Touch;
+
+static Touch finger[SCE_TOUCH_PORT_MAX_NUM][MAX_NUM_FINGERS]; // keep track of finger status
+
+typedef enum DraggingType {
+    DRAG_NONE = 0,
+    DRAG_TWO_FINGER,
+    DRAG_THREE_FINGER,
+} DraggingType;
+
+static DraggingType multi_finger_dragging[SCE_TOUCH_PORT_MAX_NUM]; // keep track whether we are currently drag-and-dropping
+
+typedef int bool;
+#define false 0;
+#define true 1;
+
+static void init_touch(void)
+{
+    for (int port = 0; port < SCE_TOUCH_PORT_MAX_NUM; port++) {
+        for (int i = 0; i < MAX_NUM_FINGERS; i++) {
+            finger[port][i].id = -1;
+        }
+        multi_finger_dragging[port] = DRAG_NONE;
+    }
+
+    for (int port = 0; port < SCE_TOUCH_PORT_MAX_NUM; port++) {
+        for (int i = 0; i < 2; i++) {
+            simulated_click_start_time[port][i] = 0;
+        }
+    }
+}
+
+void vita_handle_touch(SDL_Event *event)
+{
+    if (!touch_initialized) {
+        init_touch();
+        touch_initialized = 1;
+    }
+    preprocess_events(event);
+}
+
+static void preprocess_events(SDL_Event *event)
+{
+    // prevent suspend
+    sceKernelPowerTick(SCE_KERNEL_POWER_TICK_DISABLE_AUTO_SUSPEND);
+    sceKernelPowerTick(SCE_KERNEL_POWER_TICK_DISABLE_OLED_OFF);
+
+    // Supported touch gestures:
+    // left mouse click: single finger short tap
+    // right mouse click: second finger short tap while first finger is still down
+    // pointer motion: single finger drag
+    // left button drag and drop: dual finger drag
+    // right button drag and drop: triple finger drag
+    if (event->type == SDL_FINGERDOWN || event->type == SDL_FINGERUP || event->type == SDL_FINGERMOTION) {
+        // front (0) or back (1) panel
+        SDL_TouchID port = event->tfinger.touchId;
+        if (port < SCE_TOUCH_PORT_MAX_NUM && port >= 0) {
+            if (port == 0 || vita_rear_touch) {
+                switch (event->type) {
+                    case SDL_FINGERDOWN:
+                        preprocess_finger_down(event);
+                    break;
+                    case SDL_FINGERUP:
+                        preprocess_finger_up(event);
+                    break;
+                    case SDL_FINGERMOTION:
+                        preprocess_finger_motion(event);
+                    break;
+                }
+            }
+        }
+    }
+}
+
+static void preprocess_finger_down(SDL_Event *event)
+{
+    // front (0) or back (1) panel
+    SDL_TouchID port = event->tfinger.touchId;
+    // id (for multitouch)
+    SDL_FingerID id = event->tfinger.fingerId;
+
+    // make sure each finger is not reported down multiple times
+    for (int i = 0; i < MAX_NUM_FINGERS; i++) {
+        if (finger[port][i].id == id) {
+            finger[port][i].id = -1;
```

Comment:
Use a constant for `-1` to make it clearer what is represents.

Reply:
Fixed in 35f42ab00e2b28a49e966a752eb3016a334d53bb